src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3237,130 +3272,219 @@ private void ScanNumberAfterDot(StringBuilder sb, ref int signIndex, ref bool no
             }
         }
 
-        private static bool TryGetNumberValue(string strNum, bool hex, bool real, char suffix, long multiplier, out object result)
+        private static bool TryGetNumberValue(ReadOnlySpan<char> strNum, bool hex, bool real, NumberSuffixFlags suffix, long multiplier, out object result)
         {
             checked
             {
                 try
                 {
                     NumberStyles style = NumberStyles.AllowLeadingSign | NumberStyles.AllowDecimalPoint |
                                          NumberStyles.AllowExponent;
-                    if (suffix == 'd' || suffix == 'D')
+
+                    if (suffix == NumberSuffixFlags.Decimal)
                     {
-                        decimal d;
-                        if (Decimal.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out d))
+                        if (Decimal.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out decimal d))
                         {
                             result = d * multiplier;
                             return true;
                         }
 
+                        // Has double suffix but cannot be parsed as double.
```

Comment:
Has decimal suffix but ...

Reply:
Haha! That comment is incorrect. It wouldn't get promoted; that would fall through the the result = null; return false; statements.

I'll... quietly make sure I've caught that in my more recent work.... XD